src/Keywords/DialectKeywords.php

```diff
@@ -0,0 +1,121 @@
+ *
+ * TODO add support for dumping an example feature for a dialect directly instead.
+ *
+ * @internal
+ */
+final class DialectKeywords implements KeywordsInterface
+{
+    private GherkinDialect $currentDialect;
+
+    public function __construct(
+        private readonly DialectProviderInterface $dialectProvider,
+    ) {
+        $this->currentDialect = $this->dialectProvider->getDefaultDialect();
+    }
+
+    public function setLanguage($language): void
+    {
+        if ($language === '') {
+            throw new \InvalidArgumentException('Language cannot be empty');
+        }
+
+        $this->currentDialect = $this->dialectProvider->getDialect($language);
+    }
+
+    public function getFeatureKeywords(): string
+    {
+        return $this->getKeywordString($this->currentDialect->getFeatureKeywords());
+    }
+
+    public function getBackgroundKeywords(): string
+    {
+        return $this->getKeywordString($this->currentDialect->getBackgroundKeywords());
+    }
+
+    public function getScenarioKeywords(): string
+    {
+        return $this->getKeywordString($this->currentDialect->getScenarioKeywords());
+    }
+
+    public function getOutlineKeywords(): string
+    {
+        return $this->getKeywordString($this->currentDialect->getScenarioOutlineKeywords());
+    }
+
+    public function getExamplesKeywords(): string
+    {
+        return $this->getKeywordString($this->currentDialect->getExamplesKeywords());
+    }
+
+    public function getGivenKeywords(): string
+    {
+        return $this->getStepKeywordString($this->currentDialect->getGivenKeywords());
+    }
+
+    public function getWhenKeywords(): string
+    {
+        return $this->getStepKeywordString($this->currentDialect->getWhenKeywords());
+    }
+
+    public function getThenKeywords(): string
+    {
+        return $this->getStepKeywordString($this->currentDialect->getThenKeywords());
+    }
+
+    public function getAndKeywords(): string
+    {
+        return $this->getStepKeywordString($this->currentDialect->getAndKeywords());
+    }
+
+    public function getButKeywords(): string
+    {
+        return $this->getStepKeywordString($this->currentDialect->getButKeywords());
+    }
+
+    public function getStepKeywords(): string
+    {
+        return $this->getStepKeywordString($this->currentDialect->getStepKeywords());
+    }
+
+    /**
+     * @param list<string> $keywords
+     */
+    private function getKeywordString(array $keywords): string
+    {
+        return implode('|', $keywords);
+    }
+
+    /**
+     * @param list<string> $keywords
+     */
+    private function getStepKeywordString(array $keywords): string
+    {
+        $legacyKeywords = [];
+        foreach ($keywords as $keyword) {
+            if (str_ends_with($keyword, ' ')) {
+                $legacyKeywords[] = substr($keyword, 0, -1);
+            } else {
+                $legacyKeywords[] = $keyword . '<';
+            }
+        }
```

Comment:
If we're dumping keywords for use in example features (I think that's the only usecase?) isn't it clearer to just leave them with/without the final ` ` rather than replicating the old `<` behaviour?

Reply:
I'm replicating the old structure here to be compatible with the expectation of the KeywordsDumper (which expects the old format, but won't dump a `<` in the output)